src/ship_cmd.cpp

```diff
@@ -549,6 +587,21 @@ static void ShipController(Ship *v)
 
 	v->ShowVisualEffect();
 
+	int dz = ShipTestUpDown(v);
```

Comment:
I think a function like:
bool MoveShipUpDownOnLock(v)
would ease reading the ship controller.